src/tests/lvm/common/util_resize.go

```diff
@@ -0,0 +1,129 @@
+	}
+
+	loopDevice := e2e_agent.LoopDevice{
+		Size:   10737418240,
+		ImgDir: "/tmp",
+	}
+
+	workerNodes, err := lvm.ListLvmNode(common.NSOpenEBS())
+	Expect(err).ToNot(HaveOccurred(), "failed to list worker node")
+
+	NodeConfig = lvm.LvmNodesDevicePvVgConfig{
+		VgName:        "lvmvg",
+		NodeDeviceMap: make(map[string]e2e_agent.LoopDevice), // Properly initialize the map
+	}
+	for _, node := range workerNodes {
+		NodeConfig.NodeDeviceMap[node] = loopDevice
+	}
+
+	logf.Log.Info("setup node with loop device, pv and vg", "node config", NodeConfig)
+	err = NodeConfig.ConfigureLvmNodesWithDeviceAndVg()
+	Expect(err).ToNot(HaveOccurred(), "failed to setup node")
+
+	// setup sc parameters
+	app.Lvm = k8stest.LvmOptions{
+		VolGroup:      NodeConfig.VgName,
+		Storage:       "lvm",
+		ThinProvision: thinProvisioned,
+	}
+
+	// create sc and pvc
+	logf.Log.Info("create sc, pvc, fio pod with snapshot as source")
+	err = app.CreateVolume()
+	Expect(err).To(BeNil(), "failed to create pvc")
+
+	//verify pvc and pv to be bound
+	volUuid, err := k8stest.VerifyVolumeProvision(app.GetPvcName(), common.NSDefault)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(volUuid).ToNot(BeEmpty())
+
+	// deploy fio pod with created volume
+	logf.Log.Info("deploy fio pod with created volume")
+	err = app.DeployApplication()
+	Expect(err).To(BeNil(), "failed to deploy app")
+
+	// sleep for 30 seconds before resizing volume
+	logf.Log.Info("Sleep before resizing volume", "duration", volume_resize.DefSleepTime)
+	time.Sleep(time.Duration(volume_resize.DefSleepTime) * time.Second)
+
+	expandedVolumeSizeMb := app.VolSizeMb + 1024
+	// expand volume by editing pvc size
+	logf.Log.Info("Update volume size", "new size in MiB", expandedVolumeSizeMb)
+	_, err = k8stest.UpdatePvcSize(app.GetPvcName(), common.NSDefault, expandedVolumeSizeMb)
+	Expect(err).ToNot(HaveOccurred(), "failed to expand volume %s, error: %v", app.GetPvcName(), err)
+
+	// verify pvc capacity to new size
+	logf.Log.Info("Verify pvc resize status")
+	pvcResizeStatus, err := volume_resize.WaitForPvcResize(app.GetPvcName(), common.NSDefault, expandedVolumeSizeMb)
+	Expect(err).ToNot(HaveOccurred(), "failed to verify resized pvc %s, error: %v", app.GetPvcName(), err)
+	Expect(pvcResizeStatus).To(BeTrue(), "failed to resized pvc %s, error: %v", app.GetPvcName(), err)
+
+	// Check fio pod status
+	logf.Log.Info("Check fio pod status")
+	phase, podLogSynopsis, err := k8stest.CheckFioPodCompleted(app.GetPodName(), common.NSDefault)
+	Expect(err).To(BeNil(), "CheckPodComplete got error %s", err)
+	Expect(phase).ShouldNot(Equal(coreV1.PodFailed), "fio pod phase is %s, %s", phase, podLogSynopsis)
+
+	// wait for fio completion - monitoring log output
+	exitValue, fErr := app.WaitFioComplete(volume_resize.DefFioCompletionTime, 5)
+	Expect(fErr).ToNot(HaveOccurred())
+	logf.Log.Info("fio complete", "exit value", exitValue)
+
+	// print fio target sizes retrieved by monitoring log output
```

Comment:
What is the purpose of  this checking?

Reply:
to verify that updated size is present in monitoring log output. Should i keep it or remove it, WDY say ?

Reply:
I don't understand.
fio will have started with the initial size and then the PVC is resized. 
fio size will not change dynamically. 
If a new fio container uses the same volume (after the 1st has completed) that should use the new volume size

Reply:
got it, my mistake.
Fixed it !!